### DISCUSSION TITLE
Upload videos

### DISCUSSION THREAD

**Discussion by TodePond** [bluesky-social/atproto#1604] [Bluesky Lexicons]
**Is your feature request related to a problem? Please describe.**

I'm always frustrated when I can't upload a video of what I'm working on like I can on X, mastodon or threads.

**Describe the solution you'd like**

I want to be able to upload videos, eg: mp4 files, to posts.

**Describe alternatives you've considered**

Gifs can work for smaller clips, which is covered by this discussion: #1157 

**Additional context**

The issue #1162 was closed and we were directed towards discussion #1157 but that discussion only covers gifs, not videos. Therefore, I've opened this issue to specifically relate to videos.

**Comment by bnewbold** [comment]
In the 2x2 grid of features, this is in the corner of "high impact/importance", but also "high effort/cost".

**Reply by TodePond** [reply]
I agree with your assessment, but I think you can do it! 💪 Good luck!

**Reply by sysadminpower2019** [reply]
Yes but why would you release a platform in 2023 without video support? 
Why aren't you actually focusing on making something that works instead of selling fantasies.
Either focus on your protocol or on the bluesky implementation.
it is clear you do not have the capacity to do both. 
Also you seem to be uninterested in hiring top talent from across the global or at least outsourcing the features which you are unable to provide.
What is your ultimate goal here? Why is there no clear roadmap?
Have you planned for the next 5 steps instead of thinking only immediately in the future. 
Do you expect people to continue to use a platform that is unable provide the basic services expected from a modern platform.
why not just give up on this protocol and implement a rewrite of ActivityPub that suites the needs of bluesky.

Have you considered what will happen when people get tired of not being able to properly use your platform, get tired and move either to threads or to Masotdon or back to X. 

Who is actually in control of this project!?

**Comment by Zero3K** [comment]
I hope the ability to upload videos and display them in posts is added in the next couple of months.

**Reply by avasilic** [reply]
> I hope the ability to upload videos and display them in posts is added in the next couple of months.

Update: as of August 2024 (8 months later) there is still no support for uploading videos. Embedding videos works but uploading to Youtube or Vimeo is overcomplicating things for end users

**Reply by DavidBuchanan314** [reply]
Actual update, they're currently working on it, and it looks like the feature will ship soon-ish (although afaik no timelines have been given)

**Reply by fanpeeps** [reply]
Have managed to use client.com.atproto.repo.create_record() method to upload video blob, but post displays 'Video not found':
<img width="516" alt="Screen Shot 2024-11-15 at 3 00 56 PM" src="https://github.com/user-attachments/assets/8976d695-a259-45fc-bcc7-d687bf5eca80">

**Reply by mary-ext** [reply]
Here's some preliminary documentation around uploading videos: https://github.com/bluesky-social/bsky-docs/pull/227

**Comment by tealtxgr** [comment]
In the meantime before getting full video support, what does everyone think about being able to embed videos from, ironically, Twitter? Video support seems like a monumental task, especially at the scale of an entire social network. I have a bsky for my game I'm making, but it looks like I'll have to link to Twitter posts when new videos come out. I think that's fine in the short term though. Having native video embeds from Twitter/X would be a decent stopgap, imo.
